cumulusci/tests/pytest_plugins/pytest_sf_orgconnect.py

```diff
@@ -35,28 +84,37 @@ def project_config(runtime):
 
 
 @pytest.fixture(scope="session")
-def org_config(request, fallback_org_config):
-    """Get an org config with an active access token.
-
-    Specify the org name using the --org option when running pytest.
-    Or else it will use a dummy org.
-    """
-    org_name = sf_pytest_orgname(request)
-    if org_name:
+def cli_org_config(request):
+    """What org did the user specify on the CLI or pytest options?"""
+    cli_org_name = sf_pytest_cli_orgname(request)
+    if cli_org_name:
         with unmock_env():  # restore real homedir
             runtime = CliRuntime()
             runtime.keychain._load_orgs()
-            org_name, org_config = runtime.get_org(org_name)
+            org_name, org_config = runtime.get_org(cli_org_name)
             assert org_config.scratch, "You should only run tests against scratch orgs."
             org_config.refresh_oauth_token(runtime.keychain)
+            return org_config
+
+
+@pytest.fixture(scope="function")
+def org_config(request, current_org_shape, cli_org_config, fallback_org_config):
+    """Get an org config with an active access token.
+
+    Specify the org name using the --org option when running pytest.
+    Or else it will use a dummy org.
+    """
+    org_config = current_org_shape.org_config or cli_org_config
```

Comment:
Unless I'm misunderstanding, this appears to be opposite to the docstring: we use the current org shape if already configured, otherwise the org set on the command line, and lastly a fallback org.